pandas/core/nanops.py

```diff
@@ -1380,7 +1382,7 @@ def _maybe_null_out(
     mask: Optional[np.ndarray],
     shape: Tuple[int, ...],
     min_count: int = 1,
-) -> float:
+) -> np.ndarray | float:
```

Comment:
the return type is correct if called with a 1d array, when called with a 2d array (as is now the case for reductions) the return type is a ndarray.

Reply:
Is `|` a new notation?  im guessing this is equivalent to `Union[np.ndarray, float]`?

Reply:
since we now use `from __future__ import annotations` extensively, function annotations, both for arguments and return values, are no longer evaluated at run time and are treated as string values.

This allows us to use any notation that mypy understands.

Reply:
is that a "yes" on the `Union[np.ndarray, float]`?

Reply:
yes. 

from https://mypy-lang.blogspot.com/


> Typing Usability Improvements (PEP 585 and PEP 604)
> The necessity to repeatedly import various types and special forms from typing has been a long-term nuisance for users of static type checking and Python.
>
> Two new Python features improve this situation and are now supported by mypy:>
>
> PEP 585 lets you use list[int] instead of List[int] (no need to import List and other generic collections from typing).
> PEP 604 lets you write X | Y instead of Union[X, Y], and X | None instead of Optional[X] (no need to import Union or Optional from typing).
> Note: Using list[int] requires Python 3.9 and X | Y requires Python 3.10 (alpha) in order to work at runtime. To use them on older versions of Python, use from __future__ import annotations. This allows them to be used in type annotations, but the older variants (or string literal escaping) may be required in non-annotation contexts, such as in type aliases. See the docs for more details.

Reply:
thanks